cli/src/main.rs

```diff
@@ -23,26 +64,41 @@ pub async fn main() {
       .arg_from_usage("<APP_ID> 'Specifies the alan app to terminate'")
     )
     .subcommand(SubCommand::with_name("upgrade")
-      .about("Deploys an .agz file to an existing app in the cloud provider described in the deploy config at ~/.alan/deploy.json")
+      .about("Deploys your repository to an existing app in the cloud provider described in the deploy config at ~/.alan/deploy.json")
       .arg_from_usage("<APP_ID> 'Specifies the alan app to upgrade'")
-      .arg_from_usage("<AGZ_FILE> 'Specifies the .agz file to deploy'")
     );
 
   let matches = app.get_matches();
   match matches.subcommand() {
     ("new",  Some(matches)) => {
-      let agz_file = matches.value_of("AGZ_FILE").unwrap();
-      let cloud_alias = matches.value_of("CLOUD_ALIAS").unwrap();
-      new(agz_file, cloud_alias).await;
+      let config = get_config();
+      let cloud_alias = matches.value_of("CLOUD_ALIAS").unwrap_or(
+        config.keys().take(1).next().unwrap()
+      );
+      let body = json!({
+        "deployConfig": config,
+        "cloudAlias": cloud_alias,
+        "agzB64": anycloud_agz,
+        "DockerfileB64": get_dockerfile_b64(),
+        "appTarGzB64": get_app_tar_gz_b64(),
+      });
+      new(body).await;
     },
     ("terminate",  Some(matches)) => {
       let app_id = matches.value_of("APP_ID").unwrap();
       terminate(app_id).await;
     },
     ("upgrade",  Some(matches)) => {
+      let config = get_config();
       let app_id = matches.value_of("APP_ID").unwrap();
-      let agz_file = matches.value_of("AGZ_FILE").unwrap();
-      upgrade(app_id, agz_file).await;
+      let body = json!({
+        "appId": app_id,
+        "deployConfig": config,
+        "agzB64": anycloud_agz,
+        "DockerfileB64": get_dockerfile_b64(),
```

Comment:
nit: should camelcase take precedence over the actual name of what is described?

Reply:
I waffled on that, too. I chose to capitalize it because you *have* to capitalize the `Dockerfile` or docker ignores it.